pkg/inspector/image-inspector.go

```diff
@@ -533,16 +480,164 @@ func createOutputDir(dirName string, tempName string) (string, error) {
 		err := osMkdir(dirName, 0755)
 		if err != nil {
 			if !os.IsExist(err) {
-				return "", fmt.Errorf("Unable to create destination path: %v\n", err)
+				return "", fmt.Errorf("creating destination path: %v\n", err)
 			}
 		}
 	} else {
 		// forcing to use /var/tmp because often it's not an in-memory tmpfs
 		var err error
 		dirName, err = ioutilTempDir("/var/tmp", tempName)
 		if err != nil {
-			return "", fmt.Errorf("Unable to create temporary path: %v\n", err)
+			return "", fmt.Errorf("creating temporary path: %v\n", err)
 		}
 	}
 	return dirName, nil
 }
+
+// pullImage pulls the image using containers/image library
+// to directly connect to the docker registry
+// and will return its metadata (*types.ImageInspectInfo).
+// Assumes the image to be pulled is with a transport prefix
+// or if not attempts to add the "docker://" prefix to the image.
+func (i *defaultImageInspector) pullImage() (*types.ImageInspectInfo, digest.Digest, error) {
+	policy := &signature.Policy{
+		Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()},
+	}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, "", fmt.Errorf("creating context for policy %v: %v", policy, err)
+	}
+	defer policyContext.Destroy()
+
+	srcRef, err := alltransports.ParseImageName(i.opts.Image)
+	if err != nil {
+		// try adding "docker://"
+		i.opts.Image = "docker://" + i.opts.Image
+		srcRef, err = alltransports.ParseImageName(i.opts.Image)
+		if err != nil {
+			return nil, "", fmt.Errorf("invalid source name %s: %v", i.opts.Image, err)
+		}
+	}
+
+	certPath, err := i.certPath(i.opts.Image)
+	if err != nil {
+		return nil, "", fmt.Errorf("finding certificate path: %v", err)
+	}
+	sourceCtx := &types.SystemContext{
+		DockerCertPath: certPath,
+	}
+
+	if i.opts.DstPath, err = createOutputDir(i.opts.DstPath, "image-inspector-"); err != nil {
+		return nil, "", fmt.Errorf("creating output dir: %v", err)
+	}
+	destRef, err := directory.NewReference(i.opts.DstPath)
+	if err != nil {
+		return nil, "", fmt.Errorf("invalid destination name %s: %v", i.opts.DstPath, err)
+	}
+
+	imagePullAuths, err := i.getAuthConfigs()
+	if err != nil {
+		return nil, "", fmt.Errorf("getting registry auth config: %v", err)
+	}
+
+	// Try all the possible auths from the config file
+	log.Println("Pulling image ...")
+	for name, auth := range imagePullAuths.Configs {
+		sourceCtx.DockerAuthConfig = &types.DockerAuthConfig{
+			Username: auth.Username,
+			Password: auth.Password,
+		}
+		reportBuffer := &bytes.Buffer{}
+		err = copy.Image(policyContext, destRef, srcRef, &copy.Options{
+			RemoveSignatures: false,
+			SignBy:           "",
+			ReportWriter:     reportBuffer,
+			SourceCtx:        sourceCtx,
+			DestinationCtx:   nil,
+			ProgressInterval: PULL_LOG_INTERVAL_SEC,
+		})
+		log.Printf("Output of copy image operation using auth %s:\n%s", name, reportBuffer.String())
```

Comment:
I don't believe we can use `aggregateBytesAndReport` since `image.Copy` doesn't provide us any structured data about the transfer progress (rather it's just an output writer for logging)